util/topology/cluster.go

```diff
@@ -41,21 +41,31 @@ func GetVariable(cluster *clusterv1.Cluster, name string, value interface{}) err
 // SetMDVariable sets the variable for the given machineDeployment, overriding it if necessary, to the given value.
 // Pre-reqs: cluster.Spec.Topology != nil && cluster.Spec.Topology.Workers != nil
 func SetMDVariable(cluster *clusterv1.Cluster, mdIndex int, name string, value interface{}) error {
-	jsonValue, err := jsonValue(value)
+	jsonVal, err := jsonValue(value)
 	if err != nil {
 		return err
 	}
 
+	var varDefault *apiextensionsv1.JSON
+	switch value.(type) {
+	case []interface{}:
```

Comment:
This will not match if value is of type `[]SomeOtherType`. Looks like the only way to detect it's a slice is using reflection.